sdk-base/src/main/java/com/mapbox/maps/plugin/lifecycle/MapboxLifecyclePlugin.kt

```diff
@@ -0,0 +1,19 @@
+package com.mapbox.maps.plugin.lifecycle
+
+import android.widget.FrameLayout
+import com.mapbox.maps.MapboxLifecycleObserver
+import com.mapbox.maps.plugin.MapPlugin
+
+/**
+ * Defines interface of a MapViewPlugin.
+ */
+fun interface MapboxLifecyclePlugin : MapPlugin {
+
+  /**
+   * Register a MapboxLifecycleObserver to observe life cycle events from LifecycleOwner
+   *
+   * @param mapView the instance of mapView, will get the LifecycleOwner from mapview's parent
+   * @param observer the observer that listen to the life cycle events
+   */
+  fun registerLifecycleObserver(mapView: FrameLayout, observer: MapboxLifecycleObserver)
```

Comment:
Shall we make it generic and use `LifecycleOwner` instead of `mapView: FrameLayout`? this way it is easier to integrate with Surface as well.

Reply:
In order to get the LifecycleOwner, we must have the view firstly. However, if we want to support MapSurface, we will need to pass in a SurfaceView instance as a param for MapSurface. This will be a breaking change and example `MapWallpaper` will not works anymore since there is no view in it. I have upload a branch for MapSurface https://github.com/mapbox/mapbox-maps-android/compare/kl-lifecycle...kl-surface?expand=1
IMHO, we should not support MapSurface in this plugin.

Reply:
> In order to get the LifecycleOwner, we must have the view firstly. However, if we want to support MapSurface, we will need to pass in a SurfaceView instance as a param for MapSurface. This will be a breaking change and example MapWallpaper will not works anymore since there is no view in it. I have upload a branch for MapSurface https://github.com/mapbox/mapbox-maps-android/compare/kl-lifecycle...kl-mapsurface?expand=1
IMHO, we should not support MapSurface in this plugin.

:+1:, I do think instead of FrameLayout we can reference "View" instead.

Reply:
Yeah, we can use View here.